java/src/jmri/server/json/memory/JsonMemoryHttpService.java

```diff
@@ -50,6 +53,9 @@ public ObjectNode doGet(Memory memory, String name, String type, JsonRequest req
                 } else if (val instanceof jmri.Reporter) {
                     ObjectNode reporterValue = reporterService.doGet((jmri.Reporter) val, name, REPORTER, request);
                     data.set(VALUE, reporterValue);
+                } else if (val instanceof jmri.BasicRosterEntry) {
+                    ObjectNode rosterValue = (ObjectNode) rosterService.getRosterEntry(request.locale, ((BasicRosterEntry) val).getId(), 0);
```

Comment:
@rhwood I'd appreciate suggestions for handling the id here.

Reply:
Use `request.id` instead of 0.  `request` is a JsonRequest and will have the correct value for any context (the JsonRequest is a container  for data about the JSON message which avoids many methods needing 6 or more parameters).